### ISSUE TITLE
Type forwarding for same type defined in both System.Runtime.Interop and *.PInvoke assembly

### ISSUE THREAD

**Issue by drieseng** [dotnet/corefx#7274]
Currently, type forwarding is defined for (for example, but not limited to) the **System.Runtime.InteropServices.GuidAttribute** type in both the **System.Runtime.InteropServices.PInvoke** and the **System.Runtime.InteropServices** assembly.

The nuget packages for these assemblies are listed as dependencies of the **Microsoft.NETCore** package in the latest build (5.0.1-rc3-23925-00):

```json
      "Microsoft.NETCore/5.0.1-rc3-23925-00": {
        "type": "package",
        "dependencies": {
          "System.Runtime.InteropServices": "4.1.0-rc3-23925-00",
          "System.Runtime.InteropServices.PInvoke": "4.0.0-rc3-23925-00",
     }
```

This means that if you reference the latest build of **Microsoft.NETCore.UniversalWindowsPlatform** (5.2.0-rc3-23925-00), you'll get the following compile error whenever you use one of the types that are forwarded in these two assemblies:

*error CS0433: The type 'GuidAttribute' exists in both 'System.Runtime.InteropServices.PInvoke, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Runtime.InteropServices, Version=4.0.20.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'*

Sorry if I'm missing the obvious here ...

**Comment by joshfree** [comment]
@chcosta @ericstj

**Comment by drieseng** [comment]
I identified the problem, I had a mix of two different versions of the .NET Core nuget packages:

  "dependencies": {
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.0-rc3-23915",
    "Renci.Security.Cryptography": "1.0.0",
    "System.Xml.XPath.XmlDocument": "4.0.1-rc3-23924"
  },

(you can ignore the second dependency)

It definitely was an error on my part, sorry for wasting your time.
Once I modified the version of the **System.Xml.XPath.XmlDocument** package to *4.0.1-rc3-23915* it worked fine. I've attached the working and the failing project.json file (as .txt, as GitHub doesn't accept .json files).

I still wonder if issues like this won't be happening a lot.
If - as a library author - I reference version X of a .NET Core package, can I expect all consumers to reference the exact same version ?

[project.lock.json-failed.txt](https://github.com/dotnet/corefx/files/194678/project.lock.json-failed.txt)
[project.lock.json-working.txt](https://github.com/dotnet/corefx/files/194680/project.lock.json-working.txt)

**Comment by ericstj** [comment]
We won't be making the sorts of changes in stable builds that we are currently making in pre-release.

In this case we moved support for InteropServices.Pinvoke from netstandard1.3 to netstandard1.5.  If we had done this in a stable build that would be dropping API from an existing platform which is breaking change and not allowed.

Our general reccomendation to folks is for pre-release packages ensure that you are on a consistent version because we make no guarantees about compatibility between different pre-release builds.

**Comment by drieseng** [comment]
I just upgraded to a new nightly of the CoreFX libraries, and am now able to reproduce the issue consistently with the following steps:

* Install the .NET Core CLI (I installed the latest available, which is 1.0.0-beta-002171 at the time of writing.
* Copy a new directory locally.
* Create a project.json with the following content in that directory:
```json
{
  "dependencies": {
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.0-rc3-23931-00"
  },
  "frameworks": {
    "uap10.0": {}
  },
  "runtimes": {
    "win10-arm": {},
    "win10-arm-aot": {},
    "win10-x86": {},
    "win10-x86-aot": {},
    "win10-x64": {},
    "win10-x64-aot": {}
  }
}
```
* Create a **AssemblyInfo.cs** file with the following content in that directory:
```cs
using System.Runtime.InteropServices;
[assembly: Guid("EB0D62D0-F031-4CC3-A294-D7B867DA517B")]
```
* Start a Command Prompt session, and change to the newly created directory.
* Execute `dotnet restore && dotnet build`.

You'll get the following compile error:

error CS0433: The type 'GuidAttribute' exists in both 'System.Runtime.InteropServices.PInvoke, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Runtime.InteropServices, Version=4.0.20.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
